src/emulator/dataconnectEmulator.ts

```diff
@@ -35,21 +35,25 @@ export class DataConnectEmulator implements EmulatorInstance {
 
   async start(): Promise<void> {
     this.logger.log("DEBUG", `Using Postgres connection string: ${this.getLocalConectionString()}`);
-    const info = await DataConnectEmulator.build({ configDir: this.args.configDir });
-    if (requiresVector(info.metadata)) {
-      if (Constants.isDemoProject(this.args.projectId)) {
-        this.logger.logLabeled(
-          "WARN",
-          "Data Connect",
-          "Detected a 'demo-' project, but vector embeddings require a real project. Operations that use vector_embed will fail.",
-        );
-      } else {
-        this.logger.logLabeled(
-          "WARN",
-          "Data Connect",
-          "Operations that use vector_embed will make calls to production Vertex AI",
-        );
+    try {
+      const info = await DataConnectEmulator.build({ configDir: this.args.configDir });
+      if (requiresVector(info.metadata)) {
+        if (Constants.isDemoProject(this.args.projectId)) {
+          this.logger.logLabeled(
+            "WARN",
+            "Data Connect",
+            "Detected a 'demo-' project, but vector embeddings require a real project. Operations that use vector_embed will fail.",
+          );
+        } else {
+          this.logger.logLabeled(
+            "WARN",
+            "Data Connect",
+            "Operations that use vector_embed will make calls to production Vertex AI",
+          );
+        }
       }
+    } catch (err: any) {
```

Comment:
I don't think this try-catch is needed. 

If `fdc build` returns non zero exit code, it failed to read sources [here](https://source.corp.google.com/piper///depot/google3/third_party/firebase/dataconnect/emulator/cli/build/build.go;l=31-34;rcl=639106823).

Compile errors are still zero exit code with gql error part of output json.

Do we have a repro of this bug?

Reply:
If you look into the implementation of `.build` below, any output on stderr will be considered fatal, including verbose logs. This may be overkill. Can we agree on the desirable behavior? I think most tools have the main output (e.g. JSON) on stdout and logging should be done freely on stderr

Reply:
I agree.

We should throw err based on exit code. Stderr goes in the error message in case of failure, but otherwise debug logs

Reply:
Fair points - changed build to only throw on nonzero error codes. I do still think this should be a best effort check in this context, so I think it makes sense to keep this try/catch.